packages/ui-components/src/BrowseFileDialog.tsx

```diff
@@ -31,32 +31,80 @@ export interface IBrowseFileDialogOptions {
   multiselect?: boolean;
   includeDir?: boolean;
   acceptFileOnDblClick?: boolean;
+  rootPath?: string;
+  startPath?: string;
+}
+
+interface IBrowseFileBreadCrumbsOptions extends BreadCrumbs.IOptions {
+  rootPath?: string;
+}
+
+/**
+ * Breadcrumbs widget for browse file dialog body.
+ */
+class BrowseFileDialogBreadcrumbs extends BreadCrumbs {
+  model: any;
+  rootPath: string;
+
+  constructor(options: IBrowseFileBreadCrumbsOptions) {
+    super(options);
+    this.model = options.model;
+    this.rootPath = options.rootPath;
+  }
+
+  protected onUpdateRequest(msg: any): void {
+    super.onUpdateRequest(msg);
+    const contents = this.model.manager.services.contents;
+    const localPath = contents.localPath(this.model.path);
+
+    // if 'rootPath' is defined prevent navigating to it's parent/grandparent directories
+    if (localPath && this.rootPath && localPath.indexOf(this.rootPath) === 0) {
+      const breadcrumbs = document.querySelectorAll(
+        '.elyra-browseFileDialog .jp-BreadCrumbs > span[title]'
+      );
+
+      breadcrumbs.forEach((crumb: Element): void => {
+        if ((crumb as HTMLSpanElement).title.indexOf(this.rootPath) === 0) {
+          crumb.className = crumb.className.replace(
+            'elyra-BreadCrumbs-',
+            'jp-BreadCrumbs-'
+          );
+        } else {
+          crumb.className = crumb.className.replace(
+            'jp-BreadCrumbs-',
+            'elyra-BreadCrumbs-'
+          );
```

Comment:
I'm not sure why we need this `else` block

Reply:
the way the `BreadCrumbs` works is as you navigate, the directory nodes are actually just getting the [`textContent`](https://github.com/jupyterlab/jupyterlab/blob/master/packages/filebrowser/src/crumbs.ts#L359) and [`title`](https://github.com/jupyterlab/jupyterlab/blob/master/packages/filebrowser/src/crumbs.ts#L361) updated. therefore, since the `className` is getting changed, need to make sure to undo the changes when appropriate. otherwise the `className` would be apply to incorrect node.